huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_account_test.go

```diff
@@ -0,0 +1,147 @@
+package rds
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk/pagination"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getPgAccountResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getPgAccount: query RDS PostgreSQL account
+	var (
+		getPgAccountHttpUrl = "v3/{project_id}/instances/{instance_id}/db_user/detail?page=1&limit=100"
+		getPgAccountProduct = "rds"
+	)
+	getPgAccountClient, err := cfg.NewServiceClient(getPgAccountProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating RDS client: %s", err)
+	}
+
+	// Split instance_id and user from resource id
+	parts := strings.Split(state.Primary.ID, "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid id format, must be <instance_id>/<name>")
+	}
+	instanceId := parts[0]
+	accountName := parts[1]
+
+	getPgAccountPath := getPgAccountClient.Endpoint + getPgAccountHttpUrl
+	getPgAccountPath = strings.ReplaceAll(getPgAccountPath, "{project_id}", getPgAccountClient.ProjectID)
+	getPgAccountPath = strings.ReplaceAll(getPgAccountPath, "{instance_id}", instanceId)
+
+	getPgAccountResp, err := pagination.ListAllItems(
+		getPgAccountClient,
+		"page",
+		getPgAccountPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS PostgreSQL account: %s", err)
+	}
+
+	getPgAccountRespJson, err := json.Marshal(getPgAccountResp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS PostgreSQL account: %s", err)
+	}
+
+	var getPgAccountRespBody interface{}
+	err = json.Unmarshal(getPgAccountRespJson, &getPgAccountRespBody)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS PostgreSQL account: %s", err)
+	}
+
+	account := utils.PathSearch(fmt.Sprintf("users[?name=='%s']|[0]", accountName), getPgAccountRespBody, nil)
+
+	if account != nil {
+		return account, nil
+	}
+
+	return getPgAccountRespBody, nil
+}
+
+func TestAccPgAccount_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	updateName := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_rds_pg_account.test"
+	pwd := fmt.Sprintf("%s%s%d", acctest.RandString(5), acctest.RandStringFromCharSet(2, "!#%^*"),
```

Comment:
```suggestion
	pwd := utils.randompassword()
```
now this function has met the rule

Reply:
it has been changed to a fixed value